plugins/app-backend/src/service/router.ts

```diff
@@ -95,9 +95,28 @@ export async function createRouter(
   staticRouter.use(notFoundHandler());
 
   router.use('/static', staticRouter);
-  router.use(express.static(appDistDir));
+  router.use(
+    express.static(appDistDir, {
+      setHeaders: (res, path) => {
+        // The Cache-Control header instructs the browser to not cache html files since it might
```

Comment:
Tbh I think this is the correct behaviour either way? `index.html` is supposed to be fetched every time really and be tiny, then reference static content with hashes

Reply:
Yes. But it doesn't really work that way with the default settings of `public, max-age=0`.

I had a closer look on a Netlify deployment of ours and they generate `cache-control: public, max-age=0, must-revalidate` for the html files and a very long lasting `cache-control: public, max-age=31556926` for static files as one would expect from something like an immutable-web-app. So I'm wondering whether the `must-revalidate` is the only missing part which brings up this problem in our deployment. Maybe one of our reverse proxies (nginx -> oauth2-proxy -> backstage) doesn't follow the `max-age=0` as strictly as we need and returns an outdated cached copy.

Reply:
Thinking what we have should be fine, at least according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#preventing_caching

My initial comment was actually mainly referring to the comment, as in it does really seem like a workaround, and just the behaviour that we want.